zulipterminal/model.py

```diff
@@ -1086,6 +1088,23 @@ def _handle_update_message_flags_event(self, event: Event) -> None:
             set_count(list(message_ids_to_mark & indexed_message_ids),
                       self.controller, -1)
 
+    def _handle_update_emoji_event(self, event: Event) -> None:
+        """
+        Handle update of emoji
+        """
+        realm_emoji_update = event['realm_emoji']
+        for emoji in realm_emoji_update:
+            if realm_emoji_update[emoji]['deactivated']:
+                # Removes emojis that have been deleted
+                # from the active_emoji_data
+                _ = self.active_emoji_data.pop(
+                    realm_emoji_update[emoji]['name'], "")
+        realm_emojis = {emoji['name']: {'code': emoji_code,
+                                        'type': 'realm_emoji'}
+                        for emoji_code, emoji in realm_emoji_update.items()
+                        if not emoji['deactivated']}  # type: NamedEmojiData
```

Comment:
Note that this is functionally equivalent to `fetch_custom_emojis`, so given that function no longer actually does a fetch but rather just processes data, let's rename and reuse it.

Reply:
I have not reused it because the return value of `        mocker.patch('zulipterminal.model.Model.process_realm_emoji_data',
                     return_value=custom_emojis)` has been hardcoded. As a result my `test__handle_update_emoji_event` fails.

Reply:
See my note about parameterising the function.